$OpenBSD$

GHC 9.10+ compat: replace memory with ram in build-depends,
replace MIN_VERSION_memory CPP macro with MIN_VERSION_ram.

--- cryptostore.cabal
+++ cryptostore.cabal
@@ -62,7 +62,7 @@
   build-depends:       base >= 4.9 && < 5
                      , bytestring
                      , basement
-                     , memory
+                     , ram
   if flag(use_crypton)
     build-depends:     crypton
                      , crypton-asn1-encoding >= 0.10.0 && < 0.11
@@ -101,7 +101,7 @@
                      , X509.Tests
   build-depends:       base >= 4.9 && < 5
                      , bytestring
-                     , memory
+                     , ram
                      , tasty
                      , tasty-hunit
                      , tasty-quickcheck

--- src/Crypto/Store/Util.hs
+++ src/Crypto/Store/Util.hs
@@ -39,7 +39,7 @@
 
 -- | Reverse a bytearray.
 reverseBytes :: ByteArray ba => ba -> ba
-#if MIN_VERSION_memory(0,14,18)
+#if MIN_VERSION_ram(0,14,18)
 reverseBytes = B.reverse
 #else
 reverseBytes = B.pack . reverse . B.unpack
